lib/tree.rb

```diff
@@ -1,44 +1,77 @@
 class NoApplesError < StandardError; end
 
+# A class representing an apple tree
 class Tree
-  attr_#fill_in :height, :age, :apples, :alive
-
-  def initialize
+  MAX_AGE = 20
+  FRUIT_AGE = 2
+  GROWTH_PER_YEAR = 2 # inches
+
+  attr_accessor :height, :age, :apples, :alive
+
+  def initialize(height = 0, age = 0, apples = [], alive = true)
+    @height = height
+    @age = age
+    @apples = apples
+    @alive = alive
+    @death_age = rand(0..Tree::MAX_AGE)
   end
 
   def age!
+    if @age < @death_age
+      @age += 1
+      @height += Tree::GROWTH_PER_YEAR
+      if @age > Tree::FRUIT_AGE
+        20.times do
+          @apples << Apple.new('red', rand(1..4))
+        end
+      end
+    else
+      @alive = false
```

Comment:
It's good practice to use getters/setters when you can instead of ivars. One you may want to redefine a getter say...

```
  def apples
    @apples ||= []
  end
```